src/OrleansAzureUtils/MultiClusterNetwork/AzureTableBasedGossipChannel.cs

```diff
@@ -0,0 +1,202 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using Orleans.Runtime.Configuration;
+using Orleans.MultiCluster;
+
+namespace Orleans.Runtime.MultiClusterNetwork
+{
+    /// <summary>
+    /// An implementation of a gossip channel based on a standard Orleans Azure table.
+    /// Multiple gossip networks can use the same table, and are separated by partition key = GlobalServiceId
+    /// </summary>
+    internal class AzureTableBasedGossipChannel : IGossipChannel
+    {
+        private TraceLogger logger;
+        private GossipTableInstanceManager tableManager;
+        private static int sequenceNumber;
```

Comment:
There will usually be multiple channels. The sequence number helps to tell them apart when logging and debugging.  I copied this pattern from the storage providers.